datafusion/proto/src/to_proto.rs

```diff
@@ -455,11 +455,36 @@ impl TryFrom<&Expr> for protobuf::LogicalExprNode {
                 }
             }
             Expr::BinaryExpr(BinaryExpr { left, op, right }) => {
-                let binary_expr = Box::new(protobuf::BinaryExprNode {
-                    l: Some(Box::new(left.as_ref().try_into()?)),
-                    r: Some(Box::new(right.as_ref().try_into()?)),
+                // Try to linerize a nested binary expression tree of the same operator
```

Comment:
Is it correct that this code  will work for `(((A AND B) AND C) AND D)` but not `A AND (B AND (C AND D)))`?

Reply:
Yep, it only goes rightwards.

Reply:
Though just to clarify, `(B and (C AND D))` part of the second expression will be linearized. It just won't include `A` (the left one has a depth of 1; the right one has depth of 2).